### PR TITLE
Solana keyring mojo

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/20602

1. Fix a bug that non default keyrings being created when restoring from mnemonics
2. We can now lazily create Solana keyring when SOL account is being added (import would lazily create one too, will be addressed in different PR)
3. KeyringService now has SignMessage for non ETH use 
4. We currently use different signature for SignMessage for non-ETH keyring but we should consider [abstract HDKeyring](https://github.com/brave/brave-browser/issues/20837) because we have too many ETH logics embedded in current HDKeyring now

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by spylogsster** [comment]
Looks good!

File: components/brave_wallet/browser/keyring_service.cc [hunk 1]
```diff
@@ -626,7 +645,9 @@ HDKeyring* KeyringService::RestoreKeyring(const std::string& keyring_id,
     return nullptr;
   }
 
-  if (!CreateKeyringInternal(keyring_id, mnemonic, is_legacy_brave_wallet)) {
+  // non default keyrings can only create encryptors for lazily keyring creation
+  if (keyring_id != mojom::kDefaultKeyringId ||
```
Comment (spylogsster) [inline components/brave_wallet/browser/keyring_service.cc:649]: mind to add a test to check the keyring observers will be called only for the default keyring?

Comment (darkdh) [reply components/brave_wallet/browser/keyring_service.cc:649]: good call, fixed in cd6bbd54c7b468a741e9a7af2954bd31ac68ec58

**Review by spylogsster — APPROVED** [review]
LGTM!

**Review by bbondy — APPROVED** [review]

**Comment by darkdh** [comment]
`BraveShieldsAPIBrowserTest.AllowScriptsOnceIframe` failure is not related to this PR
https://ci.brave.com/job/pr-brave-browser-solana-keyring-mojo-macos/5/execution/node/284/log/

**Pull request merged**